### ISSUE TITLE
"Saving issues"

### ISSUE THREAD

**Issue by karpovichv** [InusualZ/iZone#1]
``` PHP
public function removeZone(Zone $zone)
{
    $this->zonesConfig->remove($zone->getName());
    $permData = $this->permConfig->getAll(false); //Doesn't this create a local copy?
    foreach($permData as $key => $value)
    {
        if(array_key_exists($zone->getName(), $value))
        {
            unset($value[$zone->getName()]);
            //$this->permConfig->setAll($permData);    ?
            return true;
        }
    }
    return false;
}
```

This method removes a zone, but doesn't remove it's permissions.

**Comment by ghost** [comment]
You have tested it?

**Comment by karpovichv** [comment]
Yes, I have tested it two times today. I used the version from this repo.
The first time I noticed it, I was testing this plugin with my friend for our server. I have added him to a zone, then removed it and created another one with the same name. He was able to break and place blocks and we thought that plugin just doesn't work, but the actual problem was that his "friend" permission wasn't removed. That time we used the version from https://forums.pocketmine.net/plugins/izone.265/ (2.7.1)

**Comment by ghost** [comment]
Well, the permission error is because the zone name not contain characters. like [qwertyuiopasdfghjklzxcvbnm]

**Comment by karpovichv** [comment]
I didn't understand you well, but if you mean that I created zones without names, not true. I created zones with at least 3 characters long names [a-z].

**Comment by ghost** [comment]
Then it is a bug.

**Comment by karpovichv** [comment]
Well, I think I figured it out. When I posted the original post, I was thinking that the problem is because

``` PHP
$permData = $this->permConfig->getAll(false);
```

creates a local copy of permission config (and it does! Double bug). But the actual problem is not so obvious.

Now, to the point. It seems like unsetting "$value[$zone->getName()]" doesn't affect the value in "$permData". And after reading the "foreach" manual at php.net, I think what it really does is destroy the reference to an array element, stored in "$value" or so...
